_examples/clone/main.go

```diff
@@ -14,11 +14,11 @@ func main() {
 	directory := os.Args[2]
 
 	// Clone the given repository to the given directory
-	Info("git clone %s %s", url, directory)
+	Info("git clone %s %s --recursive", url, directory)
 
 	r, err := git.PlainClone(directory, false, &git.CloneOptions{
-		URL:   url,
-		Depth: 1,
+		URL:               url,
+		RecurseSubmodules: git.DefaultRecursivity,
```

Comment:
I would change the name from `DefaultRecursivity` to `DefaultRecurseModules` or `DefaultSubmoduleRecursivity` or something like that, so that it does not get confused with other parameters related with recursivite of non-submodule stuff (e.g. reference resolution).